backend-plugin/src/main/java/com/redhat/jenkins/nodesharingbackend/Api.java

```diff
@@ -249,38 +251,68 @@ public void doReportWorkload(@Nonnull final StaplerRequest req, @Nonnull final S
+                    if (rs.getOwner().equals(executor)) {
+                        processed.add(rs);
+                    }
+                }
+            }
+        }
 
         final List<ReportWorkloadRequest.Workload.WorkloadItem> reportedItems = request.getWorkload().getItems();
         final ArrayList<ReservationTask> reportedTasks = new ArrayList<>(reportedItems.size());
-        final ExecutorJenkins executor = config.getJenkinsByUrl(request.getExecutorUrl());
         for (ReportWorkloadRequest.Workload.WorkloadItem item : reportedItems) {
             reportedTasks.add(new ReservationTask(executor, item.getLabel(), item.getName(), item.getId()));
         }
 
+        // Examine current queue
         Queue.withLock(new Runnable() {
-            @Override public void run() {
-                Queue queue = Jenkins.getActiveInstance().getQueue();
-                for (Queue.Item item : queue.getItems()) {
+            @Override
+            public void run() {
+                for (Queue.Item item : Jenkins.getActiveInstance().getQueue().getItems()) {
                     if (item.task instanceof ReservationTask) {
-                        // Cancel items executor is no longer interested in and keep those it cares for
-                        if (!reportedTasks.contains(item.task)) {
-                            queue.cancel(item);
+                        ReservationTask rs = (ReservationTask) item.task;
+                        // Collect what was processed already
+                        if (rs.getOwner().equals(executor)) {
+                            processed.add(rs);
+                        }
+                        // Cancel items executor is no longer interested in
+                        if (!reportedTasks.contains(rs)) {
+                            Jenkins.getActiveInstance().getQueue().cancel(rs);
+                            LOGGER.info("Item removed from the queue (no longer interested by executor) " + rs);
                         }
-                        reportedTasks.remove(item.task);
                     }
                 }
-
-                // These might have been reported just before the build started the execution on Executor so now the
-                // ReservationTask might be executing or even completed on executor, though there is no way for orchestrator
-                // to know. This situation will be handled by executor rejecting the `utilizeNode` call.
-                for (ReservationTask newTask : reportedTasks) {
-                    queue.schedule2(newTask, 0);
-                }
             }
         });
 
-        String version = this.version;
-        new ReportWorkloadResponse(pool.getConfigRepoUrl(), version).toOutputStream(rsp.getOutputStream());
+        // Remove duplicated tasks
+        for (ReservationTask rs : reportedTasks.toArray(new ReservationTask [0])) {
```

Comment:
Do we need this form just for the logging? So we do not use `reportedTasks.removeAll(processed)`...